### DISCUSSION TITLE
E1010 "more than one span in entities, blocked, missing or outside." with an easy text

### DISCUSSION THREAD

**Discussion by jeromemassot** [explosion/spaCy#11387] [Help: Coding & Implementations]
Hi spaCy users and developers,
I have the following behavior that I cannot understand.

```
doc = nlp(
    """
    Nominal Pressure: 100 Bars, Nominal Temperature: 200 K, Nominal Pressure: 30 PSI, Nominal Pressure: XX PSI, Nominal Pressure: 2000 Atm
    """
)

# pattern to capture key-value pair
pattern = [
    {"ORTH": "Nominal"}, 
    {"ORTH": {"IN": ["Pressure", "Temperature"]}}, 
    {"ORTH": ":"}, {"IS_DIGIT": True}, 
    {"LOWER": {"IN": ["bars", "psi", "k", "atm"]}}
]

def add_event_ent(matcher, doc, i, matches):
    match_id, start, end = matches[i]
    entity = Span(doc, start, end, label="SPEC")
    doc.ents += (entity,)
    print(entity.text)

matcher = Matcher(nlp.vocab)
matcher.add("Specs", [pattern], on_match=add_event_ent)

matches = matcher(doc)
```

returns : ValueError: [E1010] Unable to set entity information for token 4 which is included in more than one span in entities, blocked, missing or outside.

but the return matches have the following start-end indices:
1 6 Nominal Pressure: 100 Bars
7 12 Nominal Temperature: 200 K
13 18 Nominal Pressure: 30 PSI
25 30 Nominal Pressure: 2000 Atm

So I do not understand why I have a E1010 here.

Thanks

Best regards

Jerome

**Comment by polm** [comment]
Are you using one of the pretrained pipelines? If so that adds entities, which is where the overlap is coming from. For example:

```
import spacy
from spacy.matcher import Matcher
from spacy.tokens import Span

nlp = spacy.load("en_core_web_sm")

text = """
Nominal Pressure: 100 Bars, Nominal Temperature: 200 K, Nominal Pressure: 30 PSI, Nominal Pressure: XX PSI, Nominal Pressure: 2000 Atm
"""

doc = nlp(text)
for ent in doc.ents:
    print(ent.start, ent.end, ent.label_, ent, sep="\t")
```

Output:

```
4       5       CARDINAL        100
10      11      CARDINAL        200
13      15      ORG     Nominal Pressure
16      17      CARDINAL        30
17      18      ORG     PSI
19      21      ORG     Nominal Pressure
22      24      PERSON  XX PSI
25      27      ORG     Nominal Pressure
28      29      CARDINAL        2000
```

Couple of other things...

- if you just want to set entities based on patterns, I would recommend the EntityRuler
- when giving code examples, please provide a sample we can run (with imports etc.)
- when providing errors, please provide the full stack trace so it's clear where it's happening
